src/interface.js

```diff
@@ -0,0 +1,110 @@
+"use strict";
+
+document.addEventListener("DOMContentLoaded", () => {
+  sessionStorage.setItem("token", "");
+
+  const elements = document.getElementById("display").children;
+  const divArray = [];
+  const form = document.getElementById("userform");
+  const peepform = document.getElementById("peepform");
+
+  let peep = new Peep();
+  let peepList = new PeepList();
+  let user = new NewUser();
+  let userLogIn = new UserLogIn();
+
+  getElements();
+  showLink();
+
+  window.addEventListener("hashchange", getCurrentLink);
+
+  form.addEventListener("submit", function (event) {
+    event.preventDefault();
+    formRouter();
+  });
+
+  peepform.addEventListener("submit", function (event) {
+    event.preventDefault();
+    sendPeep();
+  });
+
+  function getCurrentLink() {
+    showLink(getLinkFromUrl(window.location)[0]);
+  }
+
+  function getLinkFromUrl(location) {
+    return location.hash.split("#"[1]);
+  }
+
+  function showLink(link = "#peeps") {
+    removeHash();
+    hideDisplayElements();
+    const element = link.substring(1);
+    revealDisplayElements(element);
+  }
+
+  function removeHash() {
+    history.pushState(
+      "",
+      document.title,
+      window.location.pathname + window.location.search
+    );
+  }
+
+  function revealDisplayElements(element) {
+    document.getElementById(element).style.display = "block";
+    if (element === "peeps") {
+      peepList.getPeeps();
+    } else if (element === "create" || element === "login") {
+      document.getElementById("siteform").style.display = "block";
+    }
+  }
+
+  function getElements() {
+    for (let i = 0; i < elements.length; i++) {
+      divArray.push(elements[i].id);
+    }
+  }
```

Comment:
This function uses variables outside itself and changes variables outside itself. We call that having side-effects. It's ok in classes, but best avoided when coding outside classes, as it's really hard to understand what changes from the outside. Instead, consider writing:
```js
const elements = document.getElementById("display").children;
const divArray = getElementIds(elements)

function getElementIds(elements) {
   const ids = []
   for (let i = 0; i < elements.length; i++) {
      ids.push(elements[i].id);
   }
   return ids
}
```
Here you can see the function itself only uses data passed in as a parameter, and doesn't change data outside itself, instead returning data, that gets assigned to a variable outside the function.
That also means you can reuse this function, or easily rename your variables.

Reply:
really useful, and is sort of similar to the mistake I was making last week - will work on this!